mono/profiler/arg-parsing.c

```diff
@@ -0,0 +1,406 @@
+#include <config.h>
+
+#include "mono-profiler-log.h"
+
+#ifdef HAVE_UNISTD_H
+#include <unistd.h>
+#endif
+
+#ifdef HAVE_SCHED_GETAFFINITY
+#include <sched.h>
+
+#  ifndef GLIBC_HAS_CPU_COUNT
+static int
+CPU_COUNT(cpu_set_t *set)
+{
+	int i, count = 0;
+
+	for (int i = 0; i < CPU_SETSIZE; i++)
+		if (CPU_ISSET(i, set))
+			count++;
+	return count;
+}
+#  endif
+#endif
+
+
+#include "mono-profiler-log.h"
+
+typedef struct {
+	const char *event_name;
+	const int mask;
+} NameAndMask;
+
+static NameAndMask event_list[] = {
+	{ "domain", DomainEvents },
+	{ "assembly", AssemblyEvents },
+	{ "module", ModuleEvents },
+	{ "class", ClassEvents },
+	{ "jit", JitCompilationEvents },
+	{ "exception", ExceptionEvents },
+	{ "gcalloc", AllocationEvents },
+	{ "gc", GCEvents },
+	{ "thread", ThreadEvents },
+	{ "calls", EnterLeaveEvents }, //calls is the old name, we can just keep it
+	//{ "inscov", InsCoverageEvents }, //this is a profiler API event, but there's no actual event for us to emit here
+	//{ "sampling", SamplingEvents }, //it makes no sense to enable/disable this event by itself
+	{ "monitor", MonitorEvents },
+	{ "gcmove", GCMoveEvents },
+	{ "gcroot", GCRootEvents },
+	{ "context", ContextEvents },
+	{ "finalization", FinalizationEvents },
+	{ "counter", CounterEvents },
+	{ "gchandle", GCHandleEvents },
+
+	{ "typesystem", TypeLoadingAlias },
```

Comment:
Hmmmmmm
Perhaps so, but that gets us to long boring names.
I rather rename it to something like types, TBH.